### PR TITLE
Release/es56 upgrade

### PR DESCRIPTION
**Pull request by MothOnMars**
This PR includes all the code & mapping changes to allow our custom indices to use Elasticsearch 5.6.

### CODE REVIEW

File: config/secrets.yml.dev [hunk 1]
```diff
@@ -78,12 +78,12 @@ dev_settings: &DEV_SETTINGS
   custom_indices:
     elasticsearch:
       reader:
-        host: http://localhost:9200
+        host: http://localhost:9256
         user: elastic
```
Comment (peggles2) [inline config/secrets.yml.dev]: going forward are we using 9256 port?

File: lib/elastic_settings.rb [hunk 2]
```diff
@@ -1,48 +1,83 @@
...
+          es_stem_filter: { type: 'stemmer', name: 'light_spanish' }
         },
         analyzer: {
           babel_analyzer: {
-            type: "custom",
-            tokenizer: "standard",
-            filter: %w(standard asciifolding lowercase) },
+            type: 'custom',
+            tokenizer: 'standard',
+            filter: %w(asciifolding lowercase) },
```
Comment (peggles2) [inline lib/elastic_settings.rb]: %w[] for arrays

File: spec/models/web_results_post_processor_spec.rb [hunk 3]
```diff
@@ -81,7 +81,7 @@
         @post_processed_results = post_processor.post_processed_results
       end
 
-      it "should replace Bing's result title with the NewsItem title" do
+      it "replaces Bing's result title with the NewsItem title" do
```
Comment (peggles2) [inline spec/models/web_results_post_processor_spec.rb]: while you'r in here make this single quote please.

File: lib/elastic_settings.rb [hunk 4]
```diff
@@ -1,48 +1,83 @@
...
+          default: {
+            type: 'custom',
+            tokenizer: 'standard',
+            filter: %w(asciifolding lowercase) },
```
Comment (peggles2) [inline lib/elastic_settings.rb]: %w[] here as well

Comment (peggles2) [reply lib/elastic_settings.rb]: and all the ones below.

File: lib/elastic_settings.rb [hunk 5]
```diff
@@ -1,48 +1,83 @@
...
-          en_synonym: { type: 'synonym', synonyms: File.readlines(Rails.root.join("config", "locales", "analysis", "en_synonyms.txt")).map(&:chomp) },
-          en_protected_filter: { type: 'keyword_marker', keywords: File.readlines(Rails.root.join("config", "locales", "analysis", "en_protwords.txt")).map(&:chomp) },
-          en_stem_filter: { type: "stemmer", name: "english" },
-          es_stop_filter: { type: "stop", stopwords: File.readlines(Rails.root.join("config", "locales", "analysis", "es_stopwords.txt")).map(&:chomp) },
-          es_protected_filter: { type: 'keyword_marker', keywords: File.readlines(Rails.root.join("config", "locales", "analysis", "es_protwords.txt")).map(&:chomp) },
-          es_synonym: { type: 'synonym', synonyms: File.readlines(Rails.root.join("config", "locales", "analysis", "es_synonyms.txt")).map(&:chomp) },
-          es_stem_filter: { type: "stemmer", name: "light_spanish" }
+          en_stop_filter: { type: 'stop', stopwords: File.readlines(Rails.root.join('config', 'locales', 'analysis', 'en_stopwords.txt')) },
```
Comment (peggles2) [inline lib/elastic_settings.rb]: These are super long. Can we break them in separate lines?

**Comment by MothOnMars** [comment]
@peggles2 , this is not yet ready for review. This will be ready once https://github.com/GSA/search-gov/pull/376 is merged.

File: spec/models/web_results_post_processor_spec.rb (hunk 3 above)
Comment (MothOnMars) [reply spec/models/web_results_post_processor_spec.rb]: This needs to be double-quoted due to the apostrophe in `Bing's`.

File: config/secrets.yml.dev (hunk 1 above)
Comment (MothOnMars) [reply config/secrets.yml.dev]: Yes - that is the port for the 5.6 cluster in CircleCi.

File: spec/models/web_results_post_processor_spec.rb (hunk 3 above)
Comment (peggles2) [reply spec/models/web_results_post_processor_spec.rb]: 👍

File: lib/elastic_settings.rb (hunk 2 above)
Comment (peggles2) [reply lib/elastic_settings.rb]: ^^^^

**Comment by MothOnMars** [comment]
Per our discussion, I'm hitting "approve" for the remaining CodeClimate failures, as they are all related to pre-existing code.

**Pull request merged**